src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.h

```diff
@@ -1450,27 +1455,27 @@ ep_rt_utf16_to_utf8_string (
     size_t len)
 {
     STATIC_CONTRACT_NOTHROW;
-
     if (!str)
         return NULL;
-    
-    // shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-    // Simple implementation to create a utf8 string from a utf16 one
-    size_t len_utf16 = len;
-    if(len_utf16 == (size_t)-1)
-        len_utf16 = ep_rt_utf16_string_len (str);
 
-    ep_char8_t *str_utf8 = reinterpret_cast<ep_char8_t *>(malloc ((len_utf16 + 1) * sizeof (ep_char8_t)));
-    if (!str_utf8)
-        return NULL;
+    ep_char8_t* lpDestStr = NULL;
+    if (static_cast<int>(len) < 0) {
+        len = 0;
+        while (str[len])
+            len++;
```

Comment:
Why the switch away from `ep_rt_utf16_string_len`?

Reply:
```suggestion
        len = ep_rt_utf16_string_len (str);
```